modules/RTC/TPCUtils.js

```diff
@@ -0,0 +1,365 @@
+import { getLogger } from 'jitsi-meet-logger';
+import transform from 'sdp-transform';
+
+import * as JitsiTrackEvents from '../../JitsiTrackEvents';
+import browser from '../browser';
+import RTCEvents from '../../service/RTC/RTCEvents';
+
+const logger = getLogger(__filename);
+const SIM_LAYER_1_RID = '1';
+const SIM_LAYER_2_RID = '2';
+const SIM_LAYER_3_RID = '3';
+
+export const SIM_LAYER_RIDS = [ SIM_LAYER_1_RID, SIM_LAYER_2_RID, SIM_LAYER_3_RID ];
+
+/**
+ * Handles track related operations on TraceablePeerConnection when browser is
+ * running in unified plan mode.
+ */
+export class TPCUtils {
+    /**
+     * @constructor
+     */
+    constructor(peerconnection) {
+        this.pc = peerconnection;
+
+        /**
+         * The simulcast encodings that will be configured on the RTCRtpSender
+         * for the video tracks in the unified plan mode.
+         */
+        this.simulcastEncodings = [
+            {
+                active: true,
+                maxBitrate: browser.isFirefox() ? 2500000 : 200000,
+                rid: SIM_LAYER_1_RID,
+                scaleResolutionDownBy: browser.isFirefox() ? 1.0 : 4.0
+            },
+            {
+                active: true,
+                maxBitrate: 700000,
+                rid: SIM_LAYER_2_RID,
+                scaleResolutionDownBy: 2.0
+            },
+            {
+                active: true,
+                maxBitrate: browser.isFirefox() ? 200000 : 2500000,
+                rid: SIM_LAYER_3_RID,
+                scaleResolutionDownBy: browser.isFirefox() ? 4.0 : 1.0
+            }
+        ];
+    }
+
+    /**
+     * Obtains local tracks for given {@link MediaType}.
+     * @param {MediaType} mediaType - audio or video.
+     * @return {Array<JitsiLocalTrack>} - array containing the local tracks
+     * attached to the peerconnection of the given media type.
+     */
+    _getLocalTracks(mediaType) {
+        const tracks = Array.from(this.pc.localTracks.values());
```

Comment:
why do we need to create thiss extra array?

Reply:
pc.localTracks is a map and therefore we need to construct an array from the interable object. I moved this to setMediaTransferActive() since there is not other function using it. I think I had one more before which I no longer needed.